src/Agent.Worker/Build/TfsVCTeeSourceProvider.cs

```diff
@@ -0,0 +1,373 @@
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Build
+{
+    public sealed class TfsVCTeeSourceProvider : SourceProvider, ISourceProvider
+    {
+        private string _workspaceName;
```

Comment:
Is there a better way to handle this? GetSourcesAsync stores value to the field, then it's used later by GetLocalPath.

In the context of thinking about syncing multiple repos, I think the best thing (from design perspective) is to pass all of the relevant variables (e.g. workspace name and sources directory) via the endpoint's data dictionary. That solves the problem of the variables dictionary (somehow correlating a variable to a specific repo endpoint).